torch_xla/csrc/tensor.cpp

```diff
@@ -284,6 +284,33 @@ void XLATensor::SetXlaData(torch::lazy::BackendDataPtr handle, bool sync) {
   }
 }
 
+void XLATensor::CreateShardedIrValue(
+    const ShardingSpecPtr sharding_spec) const {
+  torch::lazy::Value old_value = GetIrValue();
+  XLA_CHECK(old_value && old_value.node != nullptr)
+      << "Cannot create a sharded IR value if an IR value does not already "
+         "exist";
+  XLA_CHECK(old_value.node->op() == xla_device_data)
+      << "Can only set sharding for device data";
+  torch::lazy::Value new_ir = dynamic_cast<DeviceData*>(old_value.node.get())
+                                  ->CloneWithSharding(sharding_spec->sharding);
+  data()->ir_value = std::move(new_ir);
+  data()->generation += 1;
+}
+
+void XLATensor::CreateUnshardedIrValue() {
+  torch::lazy::Value old_value = CurrentIrValue();
+  XLA_CHECK(old_value && old_value.node != nullptr)
+      << "Cannot create a unsharded IR value if an IR value does not already "
+         "exist";
+  XLA_CHECK(old_value.node->op() == xla_device_data)
+      << "Can only clear sharding for device data";
+  torch::lazy::Value new_ir =
+      dynamic_cast<DeviceData*>(old_value.node.get())->Clone();
```

Comment:
Seems like we need to implement a method in each IR Node to return a unsharded IR hmm,..

Reply:
We only need to add it for the node types which will be cloned at runtime, but I'm not sure if that would be everything. We can always leave it with the base implementation (throwing a not implemented error) otherwise.